src/st/no3/3FF00.c

```diff
@@ -0,0 +1,1208 @@
+#include "no3.h"
+
+void EntityRoomTransition2(Entity* self) {
+    Entity* newEntity;
+    Primitive* prim;
+    // Used as both a loop variable and a temp
+    s32 localVar;
+
+    // These are weird fake vars needed for matching
+    Entity* gents = g_Entities;
+    Unkstruct8* layout = &g_CurrentRoomTileLayout;
+
+    if (self->ext.generic.unk80.modeS32 == 0 && self->step < 4) {
+        g_api.PlaySfx(0x631);
+        self->ext.generic.unk80.modeS32 = 0x200;
+    }
+    self->ext.generic.unk80.modeS32--;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_80180AD0);
+        layout->unk40 = 0xFC;
+        g_Player.D_80072EF4 = 0x2000;
+        D_8003C8B8 = 0;
+        g_Player.D_80072EFC = 1;
+        D_801D7DD0 |= 0x100;
+        return;
```

Comment:
```suggestion
break;

Reply:
I prefer "return" in cases like this, because when reading the code you know the function is done, rather than being the style of function where something "extra" happens after the switch statement. If tracing program flow, reaching the return means you don't need to scroll to the end of all the cases to know you've seen everything that happens for a given case. I think I will leave these as return, unless there is a competing reason not to.